### DISCUSSION TITLE
Are multiple queries with the same key but different options considered separate?

### DISCUSSION THREAD

**Discussion by sid-6581** [DamianOsipiuk/vue-query#114] [Q&A] [answered]
Sometimes I want a query to only fire once, like if I am populating a form. I don't want to keep refetching the data in the background since the initial form values will be populated only a single time. For queries like this I typically set `staleTime: Infinity`. However, other parts of my system may want to keep refetching the same thing in the background to display updated information.

So in essence, I'll have both of these on the same page:

`useQuery(queryKey)`
`useQuery(queryKey, { staleTime: Infinity })`

I didn't really see this scenario addressed in the documentation. Will this work as expected? Would the query be executed twice, or would one of the queries be able to use the cached version under the same key despite them having different options?

**Comment by aantipov** [comment, marked as answer]
The behaviour is driven by ReactQuery which is used by VueQuery under the hood.

I believe it should work correctly.

Queries are identified by key.

I think what will happen in your use case is:
1. at first, requests are deduplicated and one request is issued.
2. second hook will used the resolved value going forward (never update)
3. first hook will keep re-fetching in the background (according to default options)

Feel free to check it and tell if it works as expected :)

**Reply by sid-6581** [reply]
Thanks! I would assume it would work like that, I just wasn't sure if I was stepping into undocumented behavior territory lol. It makes sense that it's different hooks but using the same cached values underneath.

**Reply by aantipov** [reply]
I seem to be wrong there 😕 

I had a little conversation with a ReactQuery maintainer https://github.com/TkDodo/blog-comments/issues/12#issuecomment-1018525378
As I understood, as a rule of thumb, queries with the same key should have the same options, same behaviour (by abstracting them into custom hooks).

In your case if you just need to initialise the form and make sure the data is not changed, you might want to use a special query key or copy the data.

**Reply by sid-6581** [reply]
Ahh.. Glad I asked then! Thanks for the follow-up, I'll have to rethink my approach. I have all my `useQuery` calls abstracted away, but I was passing through the options, thinking that was OK. It seems like I need to be very careful about doing so.
